File: packages/algo-go/solutions/algo-00_test.go
package solutions

import "testing"

func TestMostFrequent(t *testing.T) {
	type test struct {
		expected int
		input    []int
	}
	testCases := []test{
		{1, []int{1, 3, 4, 5, 4, 1, 1}},
		{2, []int{2, 3, 4}},
	}

	for _, tt := range testCases {
		if MostFrequent(tt.input) != tt.expected {
			t.Error("Failed")
		}
	}
}

File: packages/algo-py/solutions/algo_00_test.py
import unittest
from solutions import most_frequent


class TestAlgo_00(unittest.TestCase):
    def test(self):
        input = [1, 2, 3, 2, 4, 1, 1]
        self.assertEqual(most_frequent(input), 1)


if __name__ == '__main__':
    unittest.main()
